### PR TITLE
DPDK Memory Request fixes

### PR DESCRIPTION
**Pull request by mcgov**
- add get_numa_node_count to fetch how many numa nodes are present in Lscpu
- add Free tool memory check and warnings for DPDK tests when allocating hugepages
- allocate memory per-numa node when multiple are present

### CODE REVIEW

File: lisa/tools/free.py [hunk 1]
```diff
@@ -4,20 +4,78 @@
...
+    def get_swap_size(self) -> int:
+        # Return total swap size in Mebibytes
+        return self._get_field("Swap", "total", "MiB")
+
+    def get_free_memory_mb(self) -> int:
+        return self._get_field("Mem", "free", "MiB")
 
-        raise LisaException("Failed to get swap size")
+    def get_free_memory_gb(self) -> int:
+        return self._get_field("Mem", "free", "GiB")
```
Comment (mcgov) [inline lisa/tools/free.py]: Responding to previous review question from @squirrelsc, I usually try to just wrap the tool's behavior.

Comment (mcgov) [reply lisa/tools/free.py]: Actually digging in, maybe it makes more sense to just use the default and then shift? We'll lose granularity below KiB but the behavior and style of flags has changed over time, while KibiBytes have always been the default.

Comment (mcgov) [reply lisa/tools/free.py]: yeah I'm switching it, good note thank you 🙂

**Comment by mcgov** [comment]
Noticed an issue with cores when NIC is assigned to a different numa node, working on a fix

File: lisa/tools/free.py (hunk 1 above)
Comment (squirrelsc) [reply lisa/tools/free.py]: It's ok to keep KB level. We're not wrapping the tool for general purpose use, and it's for test cases of LISA.

Comment (mcgov) [reply lisa/tools/free.py]: It was reviewing the git history that did it for me, wanting to avoid flag issues across versions

**Comment by LiliDeng** [comment]
During my testing, I always see /sys/class/net/*/device/numa_node return 0, I didn't see any regression issues bring in by this change. The logic is too complex to read, if you can add more comments in code, that would be helpful.

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**